packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.spec.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import UrlOptionsTable from './url-options-table';
+
+// Editable url options
+const urlOptions = [
+  {
+    key: 'w',
+    value: 'w-value',
+  },
+  {
+    key: 'journal',
+    value: 'j-value',
+  },
+];
+
+const connectionStringUrl = new ConnectionStringUrl(
+  'mongodb+srv://0ranges:p!neapp1es@localhost/'
+);
+urlOptions.forEach(({ key, value }) => {
+  connectionStringUrl.searchParams.set(key, value);
+});
+
+describe('UrlOptionsTable', function () {
+  let updateConnectionFormFieldSpy: sinon.SinonSpy;
+
+  describe('with SearchParams', function () {
+    beforeEach(function () {
+      updateConnectionFormFieldSpy = sinon.spy();
+      // add option that's not in Editable url options
+      connectionStringUrl.searchParams.set('readPreferences', 'primary');
+      render(
+        <UrlOptionsTable
+          updateConnectionFormField={updateConnectionFormFieldSpy}
+          connectionStringUrl={connectionStringUrl}
+        />
+      );
+    });
+
+    it('renders view correctly', function () {
+      expect(screen.getByTestId('add-url-options-button')).to.exist;
+      expect(screen.getByTestId('url-options-table')).to.exist;
+    });
+
+    it('should not render options that are not editable, but are in CS', function () {
+      expect(() => {
+        screen.getByTestId('readPreferences-table-row');
+      }).to.throw;
+      expect(() => {
+        screen.getByTestId('readPreferences-input-field');
+      }).to.throw;
+    });
+
+    // eslint-disable-next-line mocha/no-setup-in-describe
+    urlOptions.forEach(({ key, value }) => {
+      it(`renders url option in a table row - ${key}`, function () {
+        expect(screen.getByTestId(`${key}-table-row`), `${key} options row`).to
+          .exist;
+        expect(
+          screen.getByTestId(`${key}-input-field`).getAttribute('value')
+        ).to.equal(value);
+      });
+    });
+
+    it('renders new option entry with no value when user clicks on add new button', function () {
+      const button = screen.getByTestId('add-url-options-button');
+      fireEvent.click(button);
+
+      expect(screen.getByTestId('new-option-table-row')).to.exist;
+      expect(
+        screen.getByTestId('new-option-input-field').getAttribute('value')
+      ).to.equal('');
+    });
+
+    it('renders error message when user clicks twice add new button', function () {
+      const button = screen.getByTestId('add-url-options-button');
+      fireEvent.click(button);
+      fireEvent.click(button);
+      expect(screen.findByText('Please complete existing option.')).to.exist;
+    });
+
+    it('renders selected key when user select a key', function () {
+      // todo: how to test Leafy Select?
```

Comment:
I think it's probably fine not to test this one, it's getting into the internals of the select which are already tested in leafygreen.
If it's easy it might be nice to check that an option on the connection string is rendered as selected/getByText works though.